corgi/core/models.py

```diff
@@ -943,102 +922,97 @@ def latest_components(
                 )
             )
             .values_list("latest_version", flat=True)
-            .using("read_only")
+        )
+
+    def latest_components(
+        self,
+        ofuri: str,
+        model_type: str = "ProductStream",
+        include: bool = True,
+        include_inactive_streams=False,
+    ) -> "ComponentQuerySet":
+        """Return components from latest builds in a single stream."""
+
+        # the concept of 'latest component' is only relevant within product boundaries
+        # we want to constrain by product type/ofuri which is why we pass in model_type and ofuri
+        # into the get_latest_component stored proc annotation
+
+        components = (
+            self.root_components()
+            .select_related("software_build")
+            # TODO: What if model_type isn't ProductStream?
```

Comment:
Should we remove this TODO statement is prefetching productstreams is the correct and only model to prefetch here?

Reply:
dynamically set prefetch based on model_type now